src/circuit/value_commit_orchard.rs

```diff
@@ -0,0 +1,391 @@
+                commitment
+            }
+            Some(params) => {
+                // Check that magnitude is 64 bits
+                // Note: if zsa_params is not provided, this check will be performed inside the
+                // fixed-base short scalar multiplication.
+                {
+                    let lookup_config = params.sinsemilla_chip.config().lookup_config();
+                    let (magnitude_words, magnitude_extra_bits) = (6, 4);
+                    assert_eq!(
+                        magnitude_words * sinsemilla::primitives::K + magnitude_extra_bits,
+                        64
+                    );
+                    let magnitude_zs = lookup_config.copy_check(
+                        layouter.namespace(|| "magnitude lowest 60 bits"),
+                        v_net_magnitude_sign.0.clone(),
+                        magnitude_words, // 6 windows of 10 bits.
+                        false,           // Do not constrain the result here.
+                    )?;
+                    assert_eq!(magnitude_zs.len(), magnitude_words + 1);
+                    lookup_config.copy_short_check(
+                        layouter.namespace(|| "magnitude highest 4 bits"),
+                        magnitude_zs[magnitude_words].clone(),
+                        magnitude_extra_bits, // The 7th window must be a 4 bits value.
+                    )?;
+                }
+
+                // Multiply asset by magnitude, using the long scalar mul.
+                // TODO: implement a new variable base multiplication which is optimized for 64-bit scalar
+                // (the long scalar mul is optimized for pallas::Base scalar (~255-bits))
+                //
+                // magnitude_asset = [magnitude] asset
+                let magnitude_asset = {
+                    let magnitude_scalar = ScalarVar::from_base(
+                        ecc_chip.clone(),
+                        layouter.namespace(|| "magnitude"),
+                        &v_net_magnitude_sign.0,
+                    )?;
+                    let (magnitude_asset, _) = params
+                        .asset
+                        .mul(layouter.namespace(|| "[magnitude] asset"), magnitude_scalar)?;
+                    magnitude_asset
+                };
+
+                // commitment = [sign] magnitude_asset = [v_net_magnitude_sign] asset
+                magnitude_asset.mul_sign(
+                    layouter.namespace(|| "[sign] commitment"),
+                    &v_net_magnitude_sign.1,
+                )?
+            }
+        };
+
+        // blind = [rcv] ValueCommitR
+        let (blind, _rcv) = {
+            let value_commit_r = OrchardFixedBasesFull::ValueCommitR;
+            let value_commit_r = FixedPoint::from_inner(ecc_chip, value_commit_r);
+
+            // [rcv] ValueCommitR
+            value_commit_r.mul(layouter.namespace(|| "[rcv] ValueCommitR"), rcv)?
+        };
+
+        // [v] ValueCommitV + [rcv] ValueCommitR
```

Comment:
_From str4d ([2599374820](https://github.com/zcash/orchard/pull/471#discussion_r2599374820))_

```suggestion
        // [v_net_magnitude_sign] asset_base + [rcv] ValueCommitR
```